tests/bot/utils/test_messages.py

```diff
@@ -25,3 +27,86 @@ def test_sub_clyde(self):
         for username_in, username_out in test_cases:
             with self.subTest(input=username_in, expected_output=username_out):
                 self.assertEqual(messages.sub_clyde(username_in), username_out)
+
+    def test_shorten_text(self):
+        """Test all cases of text shortening by mocking messages."""
+        tests = {
+            "thisisasingleword"
+            * 10: "thisisasinglewordthisisasinglewordthisisasinglewor...",
+            "\n".join("Lets make a new line test".split()): "Lets\nmake\na...",
+            "Hello, World!"
+            * 300: (
+                "Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!"
+                "Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!"
+                "Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!Hello, World!"
+                "Hello, World!Hello, World!H..."
+            ),
+        }
+
+        for content, expected_conversion in tests.items():
+            with self.subTest(content=content, expected_conversion=expected_conversion):
+                conversion = messages.shorten_text(content)
+                self.assertEqual(conversion, expected_conversion)
+
+    def test_extract_valid_message_links(self):
+        """Test message link detection regex."""
+        test_links = (
+            "https://discord.com/channels/884793126525489203/884793128073195592/921717920869003294",
+            "http://discord.com/channels/884793126525489203/884793128073195592/921717920869003294",
+            "http://www.discord.com/channels/884793126525489203/884793128073195592/921717920869003294",
+            "http://ptb.discord.com/channels/884793126525489203/884793128073195592/921717920869003294",
+            "http://canary.discordapp.com/channels/884793126525489203/884793128073195592/921717920869003294",
+        )
+
+        for link in test_links:
+            with self.subTest(token=link):
+                results = regex.DISCORD_MESSAGE_LINK_RE.fullmatch(link)
+                self.assertIsNotNone(
+                    results, f"Valid link '{link}' was not matched by the regex"
+                )
+
+    def test_extract_invalid_message_links(self):
+        """Test message link detection regex."""
+        test_links = (
+            "https://discord.com/channels/884793126525489203/884793128073195592000/921717920869003294",
+            "http:/discord.com/channels/884793126525489203/884793128073195592/92171792086900329400",
+            "http://ptb.discord.com/channels/884793126525489203/884793128073195592000/921717920869003294",
+            "http://canary.discord.app/channels/884793126525489203/884793128073195592/921717920869003294",
+        )
```

Comment:
Currently the DISCORD_MESSAGE_LINK_RE does not match links to direct messages (for example: `https://discord.com/channels/@me/759105648322936863/911741334019264583`). We should clarify this/comment it at least here or at the regex itself. For the link unfurl feature, it's good that it doesn't match them.